src/Type/Coercer/IntCoercer.php

```diff
@@ -19,13 +21,13 @@ public function coerce($value)
             $value = (int)$value;
         }
 
-        if (!\is_int($value) || $value > PHP_INT_MAX || $value < PHP_INT_MIN) {
+        if (!\is_numeric($value) || $value > PHP_INT_MAX || $value < PHP_INT_MIN) {
             throw new InvalidTypeException(
                 \sprintf('Int cannot represent non 32-bit signed integer value: %s', $value)
             );
         }
 
-        $intValue   = (int)$value;
+        $intValue   = $value;
```

Comment:
That assignment is no longer needed, right?

Reply:
You're probably right